src/client/canvas/object/shape.ts

```diff
@@ -24,24 +24,24 @@ export type Geometry =
   | PolygonGeometry
   | TextGeometry
 
-export type ShapeOpts = {
+export type ShapeOpts<T extends Geometry> = {
   appearance: Appearance
-  geometry: Geometry
+  geometry: T
 } & GroupOpts
 
-export class Shape implements Object2d {
+export class Shape<T extends Geometry> implements Object2d {
```

Comment:
What was the reason for making this generic?

Reply:
When I use them later on as models in the SVG renders they go on the props. Having the type available there so I don't have to make new types is useful.